client/app/license/spla/add/license-spla-add.controller.js

```diff
@@ -1,53 +1,66 @@
-angular.module("Module.license").controller("LicenseSplaAddCtrl", ($scope, $translate, License, Alerter) => {
-    "use strict";
+angular.module("Module.license").controller("LicenseSplaAddCtrl", class LicenseSplaAddCtrl {
 
-    $scope.model = {
-        options: {
+    constructor ($state, $translate, Alerter, License) {
+        this.$state = $state;
+        this.$translate = $translate;
+        this.Alerter = Alerter;
+        this.License = License;
+    }
+
+    $onInit () {
+        this.options = {
             availableServers: null,
             availableTypes: null
-        },
-        selected: {
+        };
+
+        this.selected = {
             serial: null,
             type: null,
             server: null
-        }
-    };
+        };
+    }
 
-    $scope.load = () => {
-        License.splaAddAvailableServers().then(
-            (data) => {
-                $scope.model.options = data;
+    addSpla () {
+        this.License.splaAdd(this.selected.server, { serialNumber: this.selected.serial, type: this.selected.type }).then(
+            () => {
+                this.Alerter.alertFromSWS(this.$translate.instant("license_spla_add_success"), true);
             },
-            (data) => {
-                $scope.resetAction();
-                Alerter.alertFromSWS($translate.instant("license_spla_add_step1_loading_error"), data.data);
+            ({ data }) => {
+                this.Alerter.alertFromSWS(this.$translate.instant("license_spla_add_fail"), data);
             }
         );
-    };
+    }
 
-    $scope.addSpla = () => {
-        $scope.resetAction();
-        License.splaAdd($scope.model.selected.server, { serialNumber: $scope.model.selected.serial, type: $scope.model.selected.type }).then(
-            () => {
-                Alerter.alertFromSWS($translate.instant("license_spla_add_success"), true);
-            },
+    close () {
+        this.$state.go("^");
+    }
+
+    load () {
+        this.License.splaAddAvailableServers().then(
```

Comment:
A right pattern is to return all your promise, to avoid losing them and don't understand why something happens after another etc :)

Be careful with async :)